### ISSUE TITLE
Group titles not carried over into External Search

### ISSUE THREAD

**Issue by zyphlar** [doxygen/doxygen#10642]
**Describe the bug**
When using External Search, groupInternalVariableNames are used instead of Group Titles, which is confusing for users to browse since those internal identifiers are not visible anywhere else in the system

**Expected behavior**
The page (group) identifier should be the group title, and the group title should be more reliably detected (the code appears to facilitate this, but it doesn't appear to be working)

**Screenshots**

![Screenshot 2024-02-09 182038](https://github.com/doxygen/doxygen/assets/48434/b6b739e5-309a-486d-b6bf-503a5e6bc0c7)

**To Reproduce**

Create documentation for a file like this, with external search enabled. Search for "dns" and get the above screenshot surfacing the string "dnsGroup" to users, instead of the expected "group - DNS - Domain Name System" title. Changing `addtogroup` to `defgroup` doesn't seem to change anything.

```c++
/*INT_REVISION*/

/*INT_COPYRIGHT*/

/** @file dns.h
 *  @brief Domain Name Server Header File
 */

/** @addtogroup dnsGroup DNS - Domain Name System
 * The current DNS system is agnostic about the use of IPADDR, IPADDR6 or IPADDR4. The capability of C++
 * is such that you pass in any of those types and the compiler will select the correct function for
 * that type. This is a change from tools versions 3.4.0 and earlier in which explicit IPv4 and IPv6
 * functions were used.
 *
 * \note While we are pointing out the automatic functionality in this DNS API, the same is true
 * for most functions, structures and classes throughout the API libraries.
 *
 *  @{
 */

#include <nettypes.h>
#include <buffers.h>
#include <udp.h>

/** @addtogroup dnsReturnCodeGroup DNS Return Codes
 *  @{
 */
#define FOO (0)     ///< Foo
#define BAR (1)      ///< Bar

/** @} */

/** @} */
```

**Version**
1.9.8 via Windows download

**Comment by albert-github** [comment]
-Can you please also test with the current doxygen version (1.10.0)?
- Can you add the `Doxyfile` used (in a, compressed, tar or zip file!) OR EVEN BETTER
  - Can you please attach a, small, self contained example (source+configuration file in a, compressed, tar or zip file!) that allows us to reproduce the problem? Please don't add external links as they might not be persistent (also references to GitHub repositories are considered non persistent).
- can you update the image in the issue so it represents the code presented in the issue?

**Comment by albert-github** [comment]
I don't have a web server running either (I never had one running, as far as I ca remember, so it would be quite a hassle and very time consuming to get one up an running).

When I understand it correctly it is about the fact that the section titles don't appear in case of an external search whilst when 
using the, default, local search they do show as readable text in the search results.
So instead of:

![image](https://github.com/doxygen/doxygen/assets/5223533/768a75c6-37b6-4956-8417-48892b4f4ac3)


the preferred way would be (mockup in text editor):
![image](https://github.com/doxygen/doxygen/assets/5223533/5588203e-b028-4632-83fa-035225759f1b)

For the return codes we see:
```
  <doc>
    <field name="type">group</field>
    <field name="name">dnsReturnCodeGroup</field>
    <field name="url">group__dns_return_code_group.html</field>
    <field name="keywords">DNS Return Codes</field>
    <field name="text"></field>
  </doc>
```
so either the field:
```
<field name="name">dnsReturnCodeGroup</field>
```
- should read:
  ```
  <field name="name">DNS Return Codes</field>
  ```
- or an extra field should be added like:
  ```
  <field name="title">DNS Return Codes</field>
  ```

- @zyphlar  is this correct?
- @doxygen what do you think  especially regarding the content of the `<field name="name">DNS Return Codes</field>` suggestion?

**Comment by zyphlar** [comment]
@albert-github Correct. I dug into the source code a little and it seems like there's supposed to be some automatic detection where the group title gets into the name, and it seems like that just doesn't happen

**Comment by albert-github** [comment]
>  I dug into the source code a little and it seems like there's supposed to be some automatic detection where the group title gets into the name, and it seems like that just doesn't happen

Do you mean that there is a place where you saw this? If so where?

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #10653

**Comment by zyphlar** [comment]
Nice! This is the new output:

```
  <doc>
    <field name="type">group</field>
    <field name="name">DNS - Domain Name System</field>
    <field name="url">group__dns_group.html</field>
    <field name="keywords">DNS Domain Name System</field>
    <field name="text">include dns.h The current DNS system ...</field>
  </doc>
```

Could we do the same for pages as well?

### Observed

1.	page <a href="#">pageExamples::exampleShowNetworkInterfaces</a>
	... the serial port and on a web page The web page will also display...
	... clicked on to view the web page using the respective IP addresses...

```xml
  <doc>
    <field name="type">page</field>
    <field name="name">pageExamples::exampleShowNetworkInterfaces</field>
    <field name="url">example_show_network_interfaces.html</field>
    <field name="keywords"></field>
    <field name="text">Example program to display...</field>
  </doc>
```

### Expected


1.	page <a href="#">Show Network Interfaces</a>
	... the serial port and on a web page The web page will also display...
	... clicked on to view the web page using the respective IP addresses...

```xml
  <doc>
    <field name="type">page</field>
    <field name="name">Show Network Interfaces</field>
    <field name="url">example_show_network_interfaces.html</field>
    <field name="keywords"></field>
    <field name="text">Example program to display...</field>
  </doc>
```

With that tweak I think this PR is perfect!

**Comment by zyphlar** [comment]
@albert-github here's a diff for a change that worked for me:


```diff
diff --git a/src/searchindex.cpp b/src/searchindex.cpp
index aac60f07e..44041afed 100644
--- a/src/searchindex.cpp
+++ b/src/searchindex.cpp
@@ -468,6 +468,22 @@ void SearchIndexExternal::setCurrentDoc(const Definition *ctx,const QCString &an
     {
       e.args = (toMemberDef(ctx))->argsString();
     }
+    else if (ctx->definitionType()==Definition::TypeGroup)
+    {
+      const GroupDef *gd = toGroupDef(ctx);
+      if (!gd->groupTitle().isEmpty())
+      {
+        e.name = filterTitle(gd->groupTitle());
+      }
+    }
+    else if (ctx->definitionType()==Definition::TypePage)
+    {
+      const PageDef *pd = toPageDef(ctx);
+      if (pd->hasTitle())
+      {
+        e.name = filterTitle(pd->title());
+      }
+    }
     e.extId = extId;
     e.url  = url;
     it = m_docEntries.insert({key.str(),e}).first;
```

**Comment by albert-github** [comment]
@zyphlar 

Thanks for the hint and the patch, definitely makes sense.
I've added it to the proposed pull request.

**Comment by albert-github** [comment]
Code has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.11.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
